cmd/agent/app/snmpwalk.go

```diff
@@ -0,0 +1,316 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gosnmp/gosnmp"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultVersion = ""
+	defaultOID     = ""
+	defaultPort    = 161
+
+	// snmp v1 & v2c
+	defaultCommunityString = "public"
+
+	// snmp v3
+	defaultUserName      = ""
+	defaultAuthProtocol  = ""
+	defaultAuthKey       = ""
+	defaultPrivProtocol  = ""
+	defaultPrivKey       = ""
+	defaultContext       = "public"
+	defaultSecurityLevel = ""
+
+	// general communication options
+	defaultTimeout = 10 // Timeout better suited to walking
+	defaultRetries = 3
+)
+
+var (
+	// general
+	snmpVersion string
+	address     string
+	deviceIP    string
+	oid         string
+	port        uint16
+	value       uint64
+	setVersion  gosnmp.SnmpVersion
+	snmp        gosnmp.GoSNMP
+
+	// v1 & v2c
+	communityString string
+
+	// v3
+	user          string
+	authProtocol  gosnmp.SnmpV3AuthProtocol
+	authProt      string
+	authKey       string
+	privProtocol  gosnmp.SnmpV3PrivProtocol
+	privProt      string
+	privKey       string
+	snmpContext   string
+	msgFlags      gosnmp.SnmpV3MsgFlags
+	securityLevel string
+
+	// communication
+	retries int
+	timeout int
+)
+
+var (
+	snmpCmd = &cobra.Command{
+		Use:   "snmp",
+		Short: "Snmp tools",
+		Long:  ``,
+	}
+)
+
+func init() {
+
+	snmpWalkCmd.Flags().StringVarP(&snmpVersion, "snmp-version", "v", defaultVersion, "Specify SNMP version to use")
+
+	// snmp v1 or v2c specific
+	snmpWalkCmd.Flags().StringVarP(&communityString, "community-string", "C", "", "Set the community string")
+
+	// snmp v3 specific
+	snmpWalkCmd.Flags().StringVarP(&authProt, "auth-protocol", "a", defaultAuthProtocol, "Set authentication protocol (MD5|SHA|SHA-224|SHA-256|SHA-384|SHA-512)")
+	snmpWalkCmd.Flags().StringVarP(&authKey, "auth-key", "A", defaultAuthKey, "Set authentication protocol pass phrase")
+	snmpWalkCmd.Flags().StringVarP(&securityLevel, "security-level", "l", defaultSecurityLevel, "set security level (noAuthNoPriv|authNoPriv|authPriv)")
+	snmpWalkCmd.Flags().StringVarP(&snmpContext, "context", "N", defaultContext, "Set context name")
```

Comment:
I think we might be able to use lower case `-n` if we set `TraverseChildren: false` (to test, not completely sure)

```
Local Flag on Parent Commands
By default, Cobra only parses local flags on the target command, and any local flags on parent commands are ignored. By enabling Command.TraverseChildren, Cobra will parse local flags on each command before executing the target command.

command := cobra.Command{
  Use: "print [OPTIONS] [COMMANDS]",
  TraverseChildren: false,
}
```

Reply:
I looked a bit to cobra (command line lib we use) documentation  https://github.com/spf13/cobra/blob/main/user_guide.md but I'm not sure if that's possible ... :/